BLL/Services/PaymentProvider.cs

```diff
@@ -22,9 +20,9 @@ public PaymentProvider(ServiceResolver serviceAccessor)
             };
         }
 
-        public IPaymentExecute GetPaymentOperation(PaymentServiceConstants.PaymentType type)
+        public PaymentExecuteBase GetPaymentOperation(PaymentServiceConstants.PaymentType type)
         {
-            return PAYMENT_OPERATIONS[type];
+            return _paymentOperations[type];
```

Comment:
KeyNotFoundException will be thrown here in case of an invalid key? Can we have a more meaningful exception here?

Reply:
I will add NotValidPaymentOperationException